tests/dags/common/test_sensors.py

```diff
@@ -0,0 +1,197 @@
+import logging
+import unittest
+
+import pytest
+from airflow.exceptions import AirflowException
+from airflow.models import DagRun, Pool, TaskInstance
+from airflow.models.dag import DAG
+from airflow.utils.session import create_session
+from airflow.utils.state import State
+from airflow.utils.timezone import datetime
+from airflow.utils.types import DagRunType
+from common.sensors import SingleRunExternalDAGsSensor
+
+
+DEFAULT_DATE = datetime(2022, 1, 1)
+TEST_TASK_ID = "wait_task"
+
+
+@pytest.fixture(autouse=True)
+def clean_db():
+    with create_session() as session:
+        session.query(DagRun).delete()
+        session.query(TaskInstance).delete()
+
+
+def run_sensor(sensor):
+    sensor.run(start_date=DEFAULT_DATE, end_date=DEFAULT_DATE, ignore_ti_state=True)
+
+
+def create_task(dag, external_dag_ids):
+    return SingleRunExternalDAGsSensor(
+        task_id=TEST_TASK_ID,
+        external_dag_ids=[],
+        check_existence=True,
+        dag=dag,
+        pool="test_pool",
+        poke_interval=5,
+        mode="reschedule",
+    )
+
+
+def create_dag(dag_id):
+    with DAG(
+        dag_id,
+        default_args={
+            "owner": "airflow",
+            "start_date": DEFAULT_DATE,
+        },
+    ) as dag:
+        # Create a sensor task inside the DAG
+        create_task(dag, [])
+
+    return dag
+
+
+def create_dagrun(dag, dag_state):
+    return dag.create_dagrun(
+        run_id=f"{dag.dag_id}_test",
+        start_date=DEFAULT_DATE,
+        execution_date=DEFAULT_DATE,
+        state=dag_state,
+        run_type=DagRunType.MANUAL,
+    )
+
+
+class TestExternalDAGsSensor(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        Pool.create_or_update_pool("test_pool", slots=1, description="test pool")
```

Comment:
This test pool name could probably be a class variable that's used in all the tests